### PR TITLE
fix: Session Redact wrong clipping order

### PR DESCRIPTION
**Pull request by brustolin**
## :scroll: Description

Fix an error where SR clipping happens in the wrong order.

## :green_heart: How did you test it?

Unit test

## :pencil: Checklist

You have to check all boxes before merging:

- [x] I added tests to verify the changes.
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [x] I updated the docs if needed.
- [x] I updated the wizard if needed.
- [x] Review from the native team if needed.
- [x] No breaking change or entry added to the changelog.
- [x] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,7 @@
 - `SentrySdkInfo.packages` should be an array (#4626)
 - Use the same SdkInfo for envelope header and event (#4629)
 - Fixes Session replay screenshot provider crash (#4649)
+- Session Redact wrong clipping order ()
```
Comment (brustolin) [inline CHANGELOG.md:19]:
```suggestion
- Session Redact wrong clipping order (#4651)
```

**Comment by armcknight** [comment]
Is there an issue tracking this? Could we see a before/after shot from actual results in the dashboard?

**Comment by brustolin** [comment]
> Is there an issue tracking this? 
There is no issue for this, we were investigating a crash and we found the crash was caused by wrongly skipping some clips

File: Tests/SentryTests/SentryViewPhotographerTests.swift [hunk 2]
```diff
@@ -268,6 +268,32 @@ class SentryViewPhotographerTests: XCTestCase {
         ])
     }
     
+    func testLabelRedactedStackedHierarchy() throws {
```
Comment (brustolin) [inline Tests/SentryTests/SentryViewPhotographerTests.swift:271]: This test is pretty much to avoid regression. It fails for 8.42.0

**Review by philprime — APPROVED** [review]
LGTM

**Pull request merged**